tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py

```diff
@@ -0,0 +1,89 @@
+from eth2spec.test.context import default_activation_threshold, default_balances
+from eth2spec.test.helpers.genesis import create_genesis_state
+from eth2spec.test.helpers.state import transition_unsigned_block
+from eth2spec.test.helpers.block import build_empty_block_for_next_slot
+from eth2spec.test.utils import vector_test
+from eth2spec.phase0 import spec as phase0
+from eth2spec.altair import spec
+
+
+class VersionedObject(spec.Container):
+    fork_version: spec.Version
+    obj: spec.Container
+
+
+def _with_tag(tag, obj):
+    return VersionedObject(fork_version=tag, obj=obj)
+
+
+def _to_hex(data):
+    return "0x" + bytes(data).hex()
+
+
+def transition_test(pre_state_spec=phase0, 
+                    pre_version=spec.GENESIS_FORK_VERSION, 
+                    post_version=spec.ALTAIR_FORK_VERSION, 
+                    description=None, 
+                    fork_epoch=None):
+    def _wrapper(test_fn):
+        @vector_test(description)
+        def _adapter(*args, **kwargs):
+            # test gen infra passes some unnecessary arguments
+            # clear here before proceeding...
+            kwargs.pop("phase", None)
+            kwargs.pop("bls_active", None)
+
+            def pre_tag(obj):
+                return _with_tag(pre_version, obj)
+
+            def post_tag(obj):
+                return _with_tag(post_version, obj)
+
+            # the `with_state` decorators pull in more phase
+            # machinery than we need so just re-create here...
+            state = create_genesis_state(
```

Comment:
a client of the `transition_test` decorator can always build whatever `pre` state they want, i do agree it would be nice to leverage all the decorators so i'll make another pass to see if that can happen

Reply:
yeah, for sure. That's not a terrible outcome but let's see if we can reuse them first

Reply:
took a little finagling, but now we should be able to use the decorators or if not, then get a default state on the pre-fork